mypy/stubutil.py

```diff
@@ -240,11 +241,11 @@ def remove_misplaced_type_comments(source: Union[str, bytes]) -> Union[str, byte
 def common_dir_prefix(paths: List[str]) -> str:
     if not paths:
         return '.'
-    cur = os.path.dirname(paths[0])
-    for path in paths[1:]:
-        while True:
-            path = os.path.dirname(path)
-            if (cur + '/').startswith(path + '/'):
-                cur = path
-                break
-    return cur or '.'
+
+    def common_part(parts: Sequence[str]) -> bool:
+        return all(x == parts[0] for x in parts)
+
+    dirs = [os.path.dirname(p) for p in paths]
+    parts = zip(*(path.split(os.path.sep) for path in dirs))
+    common_parts = (p[0] for p in itertools.takewhile(common_part, parts))
+    return os.path.join(*common_parts) or '.'
```

Comment:
This doesn't seem to work if there is a mix of `/` and `\` path separators, which can happen on Windows.

Also, I find the new implementation to be harder to understand than the original one. To make reviewing easier, it would be better to incrementally update the original implementation. For example, apply `os.path.normpath` on all the paths before operating on them and use `os.sep` instead of `/`, but otherwise preserver the original logic.